src/commands/Admin/roleall.ts

```diff
@@ -0,0 +1,80 @@
+/*
+	Adapted from https://github.com/NotEnoughUpdates/bush-bot/blob/master/src/commands/admin/roleAll.ts
+*/
+import { FireMessage } from "@fire/lib/extensions/message";
+import { Command } from "@fire/lib/util/command";
+import 
+{ GuildMember, 
+  Role,
+  Permissions }  from 'discord.js';
+
+export default class RoleAllCommand extends Command {
+	public constructor() {
+		super('roleAll', {
+			aliases: ['role-all', 'rall'],
+			category: 'admin',
+            userPermissions: [Permissions.FLAGS.ADMINISTRATOR],
+			description: "Give a role to al the members of this server",
+			args: [
+				{
+					id: 'role',
+					type: 'role',
+					prompt: {
+						start: 'What role would you like to give to all the members of this server?',
+						retry: '{error} Pick a valid role.'
+					}
+				}
+			],
+            enableSlashCommand: true,
+            restrictTo: "guild",
+		});
+	}
+
+	async exec(
+        message: FireMessage , 
+        args: { role: Role }) {
+		if (!message.guild) return await message.util.reply('This command can only be run in a server.');
+
+		if (!message.member!.permissions.has('ADMINISTRATOR') )
+			return await message.util.reply(`You must have admin permissions to use this command.`);
+
+		if (args.role.comparePositionTo(message.guild.me!.roles.highest) >= 0 && !args.role) {
+			return await message.util.reply('I cannot add a role higher or equal to my highest role.');
+		}
+
+		let members = await message.guild.members.fetch();
```

Comment:
Ideally this would use the rest endpoint inside an iterator like [MessageIterator](https://github.com/FireDiscordBot/bot/blob/master/lib/util/iterators.ts#L29) to prevent storing the entire member list in memory for a long period of time